sdks/java/container/common.gradle

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * License); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+applyDockerNature()
+
+def imageJavaVersion = project.hasProperty('imageJavaVersion') ? project.findProperty('imageJavaVersion').replace("java", "") : '8'
+
+description = "Apache Beam :: SDKs :: Python :: Container :: Java ${imageJavaVersion} Container"
+
+configurations {
+    dockerDependency
+    sdkHarnessLauncher
+    pulledLicenses
+}
+
+dependencies {
+    pulledLicenses project(path: ":sdks:java:container", configuration: "pulledLicenses")
+    dockerDependency project(path: ":sdks:java:container", configuration: "dockerDependency")
+    sdkHarnessLauncher project(path: ":sdks:java:container", configuration: "sdkHarnessLauncher")
+}
+
+task copyDockerfileDependencies(type: Copy) {
+    from configurations.dockerDependency
+    rename "slf4j-api.*", "slf4j-api.jar"
+    rename "slf4j-jdk14.*", "slf4j-jdk14.jar"
+    rename 'beam-sdks-java-harness-.*.jar', 'beam-sdks-java-harness.jar'
+    rename 'beam-sdks-java-io-kafka.*.jar', 'beam-sdks-java-io-kafka.jar'
+    rename 'kafka-clients.*.jar', 'kafka-clients.jar'
+    into "build/target"
+}
+
+task copySdkHarnessLauncher(type: Copy) {
+    from configurations.sdkHarnessLauncher
+    into "build/target"
+}
+
+task copyPulledLicenses(type: Copy) {
+    from configurations.pulledLicenses
+    into "build/target"
+}
+
+task copyGolangLicenses(type: Copy) {
+    from "${project(':release:go-licenses:java').buildDir}/output"
+    into "build/target/go-licenses"
+    dependsOn ':release:go-licenses:java:createLicenses'
+}
+
+task skipPullLicenses(type: Exec) {
+    executable "sh"
+    args "-c", "mkdir -p build/target/third_party_licenses && touch build/target/third_party_licenses/skip"
+}
+
+docker {
+    name containerImageName(
+            name: "${project.docker_image_default_repo_prefix}java${imageJavaVersion}_sdk",
```

Comment:
Same comment about trying to keep names a bit less constructed by string append. I can see that the Java version is used in a bunch of places that need to match, so I can accept that one being injected.

How much of this could go in e.g. `applyDockerNature(...)` ?

I don't like how all of our different types of builds are defined on one monolithic file (BeamModulePlugin) with a bunch of cross-dependencies. At some point I'd love them to be factored apart into more independent plugins. But relative to the status quo, this file does seem like another kind of "nature". So this adds another kind of composition for folks to know about. In fairness, this composition is simpler and cleaner than the monolithic file. However, this is exactly how the monolithic file came to be: we had an `apply from:...` situation that grew and grew until it was impossible to deal with.

Reply:
If I adapted it to applyDockerNature I'd probably do it in a different PR since I'll be potentially affecting all the other containers